slither/solc_parsing/declarations/using_for_top_level.py

```diff
@@ -0,0 +1,152 @@
+"""
+    Using For Top Level module
+"""
+import logging
+from typing import TYPE_CHECKING, Dict, Union, Any
+
+from slither.core.compilation_unit import SlitherCompilationUnit
+from slither.core.declarations.using_for_top_level import UsingForTopLevel
+from slither.core.scope.scope import FileScope
+from slither.core.solidity_types import TypeAliasTopLevel
+from slither.core.declarations import (
+    StructureTopLevel,
+    EnumTopLevel,
+)
+from slither.solc_parsing.declarations.caller_context import CallerContextExpression
+from slither.solc_parsing.solidity_types.type_parsing import parse_type
+from slither.core.solidity_types.user_defined_type import UserDefinedType
+
+if TYPE_CHECKING:
+    from slither.solc_parsing.slither_compilation_unit_solc import SlitherCompilationUnitSolc
+
+LOGGER = logging.getLogger("UsingForTopLevelSolc")
+
+
+class UsingForTopLevelSolc(CallerContextExpression):  # pylint: disable=too-few-public-methods
+    """
+    UsingFor class
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        uftl: UsingForTopLevel,
+        top_level_data: Dict,
+        slither_parser: "SlitherCompilationUnitSolc",
+    ):
+        self._type_name = top_level_data["typeName"]
+        self._global = top_level_data["global"]
+
+        if "libraryName" in top_level_data:
+            self._library_name = top_level_data["libraryName"]
+        else:
+            self._functions = top_level_data["functionList"]
+
+        self._using_for = uftl
+        self._slither_parser = slither_parser
+
+    def analyze(self) -> None:
+        type_name = parse_type(self._type_name, self)
+        self._using_for.using_for[type_name] = []
+
+        if hasattr(self, "_library_name"):
```

Comment:
I think we should set `_library_name` to None, and compare against it. It's a better practice to always init the variables in `__init__` rather than to use `hasattr`